speechbrain/utils/recipe_tests.py

```diff
@@ -0,0 +1,453 @@
+            )
+            return False
+        var_value = float(var_value.group(1))
+        check = check_threshold(threshold, var_value)
+
+        if not (check):
+            print(
+                "\tERROR: The variable %s of file %s (recipe %s) violated the specified threshold (%s %s)"
+                % (variable, filename, recipe_id, var_value, threshold)
+            )
+
+        break
+
+    return check
+
+
+def check_threshold(threshold, value):
+    """Checks if the value satisfied the threshold constraints.
+
+    Arguments
+    ---------
+    threshold: str
+        String that contains the contains. E.g, ">=10" or "==15" or "<5".
+    value: float
+        Float corresponding to the value to test
+
+    Returns
+    ---------
+    bool
+        True if the constraint is satisfied, False otherwise.
+    """
+
+    # Get threshold value:
+    th_value = float(
+        threshold.strip().replace("=", "").replace(">", "").replace("<", "")
+    )
+
+    # Check Threshold
+    if "==" in threshold:
+        return value == th_value
+
+    elif ">=" in threshold:
+        return value >= th_value
+
+    elif ">" in threshold:
+        return value > th_value
+
+    elif "<=" in threshold:
+        return value <= th_value
+
+    elif "<" in threshold:
+        return value < th_value
+    else:
+        return False
+
+
+def run_test_cmd(cmd, stdout_file, stderr_file):
+    """Run the command corresponding to a recipe test. The standard output and
+    the standard error is saved in the specified paths.
+
+    Arguments
+    ---------
+    cmd: str
+        String corresponding to the command to run.
+    stdout_file: path
+        File where standard output is stored.
+    stderr_file: path
+        File where standard error is stored.
+
+    Returns
+    ---------
+    rc: bool
+        The return code obtained after running the command. If 0, the test is
+        run without errors. If >0 the execution failed.
+    """
+    f_stdout = open(stdout_file, "w")
+    f_stderr = open(stderr_file, "w")
+    child = sp.Popen([cmd], stdout=f_stdout, stderr=f_stderr, shell=True)
+    child.communicate()[0]
+    rc = child.returncode
+    f_stdout.close()
+    f_stderr.close()
+    return rc
+
+
+def create_folder(folder):
+    """Run the command corresponding to a recipe test. The standard output and
+    the standard error is saved in the specified paths.
+
+    Arguments
+    ---------
+    cmd: str
+        String corresponding to the command to run.
+    stdout_file: path
+        File where standard output is stored.
+    stderr_file: path
+        File where standard error is stored.
+
+    Returns
+    ---------
+    rc: bool
+        The return code obtained after running the command. If 0, the test is
+        run without errors. If >0 the execution failed.
+    """
+    try:
+        os.makedirs(folder)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
```

Comment:
Is this function necessary? I think `os.makedirs` already includes an option called `exist_ok` which ignores if folder already exists.